worlds/witness/__init__.py

```diff
@@ -69,13 +70,9 @@ def generate_basic(self):
         pool.remove(items_by_name[random_good_item])
 
         # Put in junk items to fill the rest
-        junk_pool = self.items.JUNK_WEIGHTS.copy()
-        junk_pool = self.world.random.choices(
-            list(junk_pool.keys()), weights=list(junk_pool.values()),
-            k=len(self.locat.CHECK_LOCATION_TABLE) - len(pool) - len(self.locat.EVENT_LOCATION_TABLE) - 1
-        )
+        junk_size = len(self.locat.CHECK_LOCATION_TABLE) - len(pool) - len(self.locat.EVENT_LOCATION_TABLE) - 1
 
-        pool += [self.create_item(junk) for junk in junk_pool]
+        pool += [self.create_item(junk) for junk in weighted_list(self.items.JUNK_WEIGHTS, junk_size)]
```

Comment:
I think a change simular to this should also be applied to the get_filler_item_name method

Reply:
This seems kinda hard, tbh... The new deterministic algorithm really isn't suited to generating one junk item at a time. But I have an idea of how I could pull it off...

Reply:
Well, I have a way of doing it. But I think it'll make the code less efficient. Creating one item at a time to match some sort of condition is super slow because I have to check the condition every time